File: src/INV/CommonBundle/Entity/Revision.php
<?php

namespace INV\CommonBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Revision
 *
 * @ORM\Table(name="revision")
 * @ORM\Entity
 */
class Revision {
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="bigint", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var bool
     *
     * @ORM\Column(name="activo", type="boolean", nullable=false)
     */
    private $activo;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="fecha_en_sistema", type="date", nullable=false)
     */
    private $fechaEnSistema;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="fecha_excel", type="date", nullable=false)
     */
    private $fechaExcel;

    /**
     * @var string
     *
     * @ORM\Column(name="excel_url", type="string", length=500, nullable=false)
     */
    private $excelUrl;


    /**
     * Get id.
     *
     * @return int
     */
    public function getId() {
        return $this->id;
    }

    /**
     * Set activo.
     *
     * @param bool $activo
     *
     * @return Revision
     */
    public function setActivo($activo) {
        $this->activo = $activo;

        return $this;
    }

    /**
     * Get activo.
     *
     * @return bool
     */
    public function getActivo() {
        return $this->activo;
    }

    /**
     * Set fechaEnSistema.
     *
     * @param \DateTime $fechaEnSistema
     *
     * @return Revision
     */
    public function setFechaEnSistema($fechaEnSistema) {
        $this->fechaEnSistema = $fechaEnSistema;

        return $this;
    }

    /**
     * Get fechaEnSistema.
     *
     * @return \DateTime
     */
    public function getFechaEnSistema() {
        return $this->fechaEnSistema;
    }

    /**
     * Set fechaExcel.
     *
     * @param \DateTime $fechaExcel
     *
     * @return Revision
     */
    public function setFechaExcel($fechaExcel) {
        $this->fechaExcel = $fechaExcel;

        return $this;
    }

    /**
     * Get fechaExcel.
     *
     * @return \DateTime
     */
    public function getFechaExcel() {
        return $this->fechaExcel;
    }

    /**
     * Set excelUrl.
     *
     * @param string $excelUrl
     *
     * @return Revision
     */
    public function setExcelUrl($excelUrl) {
        $this->excelUrl = $excelUrl;

        return $this;
    }

    /**
     * Get excelUrl.
     *
     * @return string
     */
    public function getExcelUrl() {
        return $this->excelUrl;
    }
}

File: bd/java/inv-dao/src/cu/cenpis/gps/inv/data/entity/Revision.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cu.cenpis.gps.inv.data.entity;

import java.io.Serializable;
import java.util.Date;
import java.util.List;
import javax.persistence.Basic;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQueries;
import javax.persistence.NamedQuery;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.bind.annotation.XmlTransient;

/**
 *
 * @author farias-i5
 */
@Entity
@Table(schema = "")
@XmlRootElement
@NamedQueries({
    @NamedQuery(name = "Revision.findAll", query = "SELECT r FROM Revision r"),
    @NamedQuery(name = "Revision.findByIdRevision", query = "SELECT r FROM Revision r WHERE r.idRevision = :idRevision"),
    @NamedQuery(name = "Revision.findByActivo", query = "SELECT r FROM Revision r WHERE r.activo = :activo"),
    @NamedQuery(name = "Revision.findByFechaEnSistema", query = "SELECT r FROM Revision r WHERE r.fechaEnSistema = :fechaEnSistema"),
    @NamedQuery(name = "Revision.findByFechaExcel", query = "SELECT r FROM Revision r WHERE r.fechaExcel = :fechaExcel"),
    @NamedQuery(name = "Revision.findByExcelUrl", query = "SELECT r FROM Revision r WHERE r.excelUrl = :excelUrl")})
public class Revision implements Serializable {
    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Basic(optional = false)
    @Column(name = "id_revision")
    private Long idRevision;
    @Basic(optional = false)
    @NotNull
    private Boolean activo;
    @Basic(optional = false)
    @NotNull
    @Column(name = "fecha_en_sistema")
    @Temporal(TemporalType.DATE)
    private Date fechaEnSistema;
    @Basic(optional = false)
    @NotNull
    @Column(name = "fecha_excel")
    @Temporal(TemporalType.DATE)
    private Date fechaExcel;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 500)
    @Column(name = "excel_url")
    private String excelUrl;
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "revision")
    private List<Metadata> metadataList;
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "revision")
    private List<ActivoFijo> activoFijoList;

    public Revision() {
    }

    public Revision(Boolean activo, Date fechaEnSistema, Date fechaExcel, String excelUrl) {
        this.activo = activo;
        this.fechaEnSistema = fechaEnSistema;
        this.fechaExcel = fechaExcel;
        this.excelUrl = excelUrl;
    }

    public Long getIdRevision() {
        return idRevision;
    }

    public void setIdRevision(Long idRevision) {
        this.idRevision = idRevision;
    }

    public Boolean getActivo() {
        return activo;
    }

    public void setActivo(Boolean activo) {
        this.activo = activo;
    }

    public Date getFechaEnSistema() {
        return fechaEnSistema;
    }

    public void setFechaEnSistema(Date fechaEnSistema) {
        this.fechaEnSistema = fechaEnSistema;
    }

    public Date getFechaExcel() {
        return fechaExcel;
    }

    public void setFechaExcel(Date fechaExcel) {
        this.fechaExcel = fechaExcel;
    }

    public String getExcelUrl() {
        return excelUrl;
    }

    public void setExcelUrl(String excelUrl) {
        this.excelUrl = excelUrl;
    }

    @XmlTransient
    public List<Metadata> getMetadataList() {
        return metadataList;
    }

    public void setMetadataList(List<Metadata> metadataList) {
        this.metadataList = metadataList;
    }

    @XmlTransient
    public List<ActivoFijo> getActivoFijoList() {
        return activoFijoList;
    }

    public void setActivoFijoList(List<ActivoFijo> activoFijoList) {
        this.activoFijoList = activoFijoList;
    }

    @Override
    public int hashCode() {
        int hash = 0;
        hash += (idRevision != null ? idRevision.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object object) {
        // TODO: Warning - this method won't work in the case the id fields are not set
        if (!(object instanceof Revision)) {
            return false;
        }
        Revision other = (Revision) object;
        if ((this.idRevision == null && other.idRevision != null) || (this.idRevision != null && !this.idRevision.equals(other.idRevision))) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "cu.Revision[ idRevision=" + idRevision + " ]";
    }
    
}
